common/utils/include/claragenomics/utils/device_buffer.cuh

```diff
@@ -48,7 +48,7 @@ public:
 
     ~device_buffer()
     {
-        CGA_CU_CHECK_ERR(cudaFree(data_));
```

Comment:
Why is this check not necessary?

Reply:
This check would throw an exception in the destructor. Removing this check should just postpone the error detection to the next API call.

Reply:
Does this answer your question?

Reply:
Yes, that makes sense. It will make it slightly harder to debug the code, but it's better than throwing in the destructor.